pkg/cluster/nodeutils/util.go

```diff
@@ -76,13 +78,37 @@ func CopyNodeToNode(a, b nodes.Node, file string) error {
 
 // LoadImageArchive loads image onto the node, where image is a Reader over an image archive
 func LoadImageArchive(n nodes.Node, image io.Reader) error {
-	cmd := n.Command("ctr", "--namespace=k8s.io", "images", "import", "-").SetStdin(image)
+	snapshotter, err := getSnapshotter(n)
+	if err != nil {
+		return err
+	}
+	cmd := n.Command("ctr", "--namespace=k8s.io", "images", "import", "--snapshotter", snapshotter, "-").SetStdin(image)
```

Comment:
I think  it will be more robust if we don't use the snaphotter flag if we fail to find it or there is an error parsing the conffig file, instead of returning

Reply:
We fail if we can't read the snapshotter because that shouldn't happen and that's an error that will cause subtle issues if we don't pass the snapshotter.

Reply:
well, my concern is that before we didn't and I really don't know if is valid not having a snapshotter defined :smile:

Reply:
It's valid but you get the default and if the CRI is not on the default this is a problem and we get into a confusing failure mode 🙃

Not being able to detect the snapshotter either means we failed to exec, and therefore load will fail anyhow, or parsing is broken (which shouldn't happen anytime soon, the containerd config is versioned and kind explicitly used V2 for some time now). In the future this method might need to handle multiple versions of config but we can parse that first and then select the correct field.